src/twisted/protocols/_smb/ismb.py

```diff
@@ -0,0 +1,61 @@
+# -*- test-case-name: twisted.protocols._smb.tests -*-
+# Copyright (c) Twisted Matrix Laboratories.
+# See LICENSE for details.
+"""
+Various interfaces for realms, avatars and related objects
+"""
+
+from typing import List, Union
+
+from zope.interface import Attribute, Interface
+
+
+class NoSuchShare(Exception):
+    pass
+
+
+class ISMBServer(Interface):
+    """
+    A SMB server avatar, contains a number of "shares" (filesystems/printers/
+    IPC pipes)
+    """
+
+    session_id = Attribute("the assigned int64 session ID")
```

Comment:
We can clean up some of the coding-standard issues in the internals after the fact, but public interfaces really need to have names we can live with long-term :)

```suggestion
    sessionID = Attribute("the assigned int64 session ID")
```